newIDE/app/src/VariablesList/index.js

```diff
@@ -149,22 +174,40 @@ export default class VariablesList extends React.Component<Props, State> {
           variable,
           depth + 1,
           index,
-          parentVariable
+          parentVariable,
+          origin
         );
       })
     );
   }
 
+  _getVariableOrigin = (name: string) => {
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
+    return inheritedVariablesContainer // We check for 3 types of variable states, when editing instance variables
+      ? variablesContainer.has(name) && !inheritedVariablesContainer.has(name)
+        ? '' // -variable that is unique to the instance - we can edit its name/structure
+        : !variablesContainer.has(name) && inheritedVariablesContainer.has(name)
+        ? 'parent' // -displaying parent variable at its default value, changing it will create an inherited
+        : variablesContainer.has(name) && inheritedVariablesContainer.has(name)
+        ? 'inherited' // -instance variable created when editing a parent variable's value, deleting it will re-show
+        : '' // parent go in inheritedVariablesContainer, the rest stay in variablesContainer
+      : '';
+  };
+
   _renderVariableAndChildrenRows(
     name: string,
     variable: gdVariable,
     depth: number,
     index: number,
-    parentVariable: ?gdVariable
+    parentVariable: ?gdVariable,
+    parentOrigin: ?string = null
```

Comment:
`parentOrigin: 'parent' | 'inherited' | '' | null,`

Actually, you can define a type `export type VariableOrigin = 'parent' | 'inherited' | ''` and import it where needed. (and here write: `parentOrigin: ?VariableOrigin` - super readable :))